File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (!head || k == 0) return head;
        
        int size = 1;
        ListNode* ptr = head;

        while(ptr->next){
            ptr = ptr->next;
            size ++;
        }
        k = k % size ;
        int k2 = size - k;
        cout<<k2<<endl;

        ptr->next = head;
        while(k2--){
            ptr = ptr->next;
        }
        ListNode* nhead = ptr->next;
        ptr->next = nullptr ;

        return nhead;

    }
};

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or k == 0:
            return head

        
        size = 1
        ptr = head
        while ptr.next:
            ptr = ptr.next
            size += 1

        
        k = k % size
        k2 = size - k

        
        ptr.next = head

        
        while k2 > 0:
            ptr = ptr.next
            k2 -= 1

        
        new_head = ptr.next
        ptr.next = None

        return new_head
